touchforms/backend/persistence.py

```diff
@@ -24,32 +26,132 @@ def restore(sess_id, factory, override_state=None):
         state.update(override_state)
     return factory(**state)
 
-# TODO integrate with real caching framework (django ideally)
-def cache_set(key, value, timeout):
-    with open(cache_path(key), 'w') as f:
-        f.write(json.dumps(value).encode('utf8'))
 
-# TODO integrate with real caching framework (django ideally)
+def cache_set(key, value):
+    if key is None:
+        raise KeyError
+    if settings.USES_POSTGRES:
+        postgres_set(key, value)
+    else:
+        with open(cache_get_file_path(key), 'w') as f:
+            f.write(json.dumps(value).encode('utf8'))
+
+
 def cache_get(key):
+    if key is None:
+        raise KeyError
+    if settings.USES_POSTGRES:
+        try:
+            return postgres_lookup(key)
+        except KeyError:
+            return cache_get_file(key)
+    else:
+        return cache_get_file(key)
+
+
+def postgres_select(cursor, key):
+    sel_sql = replace_table("SELECT * FROM %(kwarg)s WHERE sess_id=?")
+    sel_params = [str(key)]
+    cursor.execute(sel_sql, sel_params)
+
+
+def postgres_delete(cursor, key):
+    postgres_select(cursor, key)
+    if cursor.rowcount is not 0:
```

Comment:
Curious why you're selecting/counting before deleting? Two queries is almost certainly slower than the delete alone, even if there is nothing there to delete.

Also the idiomatic way to test for numeric equality is with the `==` / `!=` operators rather than `is` / `is not`.